### DISCUSSION TITLE
ResizeObserver

### DISCUSSION THREAD

**Discussion by jonasgroendahl** [vercel/next.js#17618] [Help] [answered]
I'm trying to use a library that uses ResizeObserver API. Specifically it's a React select component library. Everytime I refresh the page, I get the following error:

<img width="983" alt="Screen Shot 2020-10-05 at 10 30 42 AM" src="https://user-images.githubusercontent.com/23399953/95057113-fdd0b600-06f5-11eb-883e-963d58d6d248.png">

How can I fix this issue in Nextjs? I suppose the problem this API is not available on the server but I couldn't find any docs that explains what to do

**Comment by lfades** [comment, marked as answer]
As you said, [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) is a browser API that is not available in Node.js, so when Next.js tries to server-render the page it will fail.

The solution here is to use [`next/dynamic` and load the component only in the browser](https://nextjs.org/docs/advanced-features/dynamic-import#with-no-ssr) 👍 .

**Reply by jonasgroendahl** [reply]
Thanks!
